### PR TITLE
WFLY-18457 Avoid creating duplicate thread groups on server reload

### PR DESCRIPTION
**Pull request by TomasHofman**
https://issues.redhat.com/browse/WFLY-18457

### CODE REVIEW

File: clustering/ee/infinispan/src/main/java/org/wildfly/clustering/ee/infinispan/scheduler/SchedulerTopologyChangeListener.java [hunk 1]
```diff
@@ -58,8 +58,11 @@
 @Listener
 public class SchedulerTopologyChangeListener<I, K extends Key<I>, V> implements ListenerRegistrar {
 
+    // Reuse this static thread group instead of creating new group every time this class is initialized.
+    private static final ThreadGroup THREAD_GROUP = new ThreadGroup(SchedulerTopologyChangeListener.class.getSimpleName());
+
```
Comment (pferraro) [inline clustering/ee/infinispan/src/main/java/org/wildfly/clustering/ee/infinispan/scheduler/SchedulerTopologyChangeListener.java:63]: Creating a ThreadGroup requires a permissions check.

File: weld/subsystem/src/main/java/org/jboss/as/weld/services/bootstrap/WeldExecutorServices.java [hunk 2]
```diff
@@ -170,4 +169,9 @@ public void run() {
             runnable.run();
         }
     }
+
+    // Holder class to postpone thread group creation until when it's needed.
+    private static class ThreadGroupHolder {
+        private static final ThreadGroup THREAD_GROUP = new ThreadGroup("Weld ThreadGroup");
+    }
```
Comment (pferraro) [inline weld/subsystem/src/main/java/org/jboss/as/weld/services/bootstrap/WeldExecutorServices.java:176]: Creating a ThreadGroup requires a permissions check.

**Review by pferraro — CHANGES_REQUESTED** [review]

**Comment by bstansberry** [comment]
@TomasHofman @pferraro @manovotn

Note that https://github.com/wildfly/wildfly/pull/17193 conflicts with this.

This touches on cross-cutting topics that we may want to standardize across the wildfly/wildfly and wildfly/wildfly-core code bases, so to increase visibility I posted https://lists.jboss.org/archives/list/wildfly-dev@lists.jboss.org/thread/2NICNYJE4NO2HK67DOFTPZUGNAKD5RRM/

My intent there isn't to make a lot of trouble; it's more that if we are going to be touching these it makes sense to think about all aspects of how we use them.

**Comment by TomasHofman** [comment]
> Note that https://github.com/wildfly/wildfly/pull/17193 conflicts with this.

I'm going to update this PR to resolve the conflict, if we decide to proceed.

**Comment by manovotn** [comment]
> Note that #17193 conflicts with this.

I have updated the above PR to link it to the WFLY issue and format accordingly.
The previous CI run didn't show any problems removing the thread group from Weld's subsystem.

**Comment by TomasHofman** [comment]
Closing in favor of https://github.com/wildfly/wildfly/pull/17194, thanks!

**Pull request closed without merging**